### PR TITLE
 api: declare our directory structure and naming an api, use it 

### PR DESCRIPTION
**Pull request by stevekuznetsov**
config: rename type storing data by filename

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

config: use Metadata type when loading from directories

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

api: declare our directory structure and naming an api, use it

We have an API/interface around how we format names and identifiers
based on what org/repo/branch/variant is being considered. Refactor code
to make this an obvious API.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

/assign @openshift/openshift-team-developer-productivity-test-platform

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
```
 2020/05/05 18:48:07 Container test in pod unit completed successfully
2020/05/05 18:48:07 error: unable to signal to artifacts container to terminate in pod unit, triggering deletion: could not run remote command: unable to upgrade connection: container not found ("artifacts")
2020/05/05 18:48:07 error: unable to retrieve artifacts from pod unit: could not read gzipped artifacts: unable to upgrade connection: container not found ("artifacts")
2020/05/05 18:49:08 No custom metadata found and prow metadata already exists. Not updating the metadata.
2020/05/05 18:49:09 Ran for 8m4s
error: some steps failed:
  * could not run steps: step unit failed: test "unit" failed: the pod ci-op-wwx3i8f2/unit was deleted without completing after 3m32s (failed containers: ) 
```

weird

/retest

File: cmd/ci-operator-configresolver/main.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ import (
 	"errors"
 	"flag"
 	"fmt"
+	"github.com/openshift/ci-tools/pkg/api"
```
Comment (alvaroaleman) [inline cmd/ci-operator-configresolver/main.go:8]: ...
Im going to look into that linter.

Comment (stevekuznetsov) [reply cmd/ci-operator-configresolver/main.go:8]: I don't have the motivation to fix it :(

**Comment by bbguimaraes** [comment]
LGTM except for the import ordering.

**Pull request merged**